bld/CLMBuildNamelist.pm

```diff
@@ -3960,50 +3960,65 @@ sub setup_logic_fire_emis {
 #-------------------------------------------------------------------------------
 
 sub setup_logic_dust_emis {
-  # Logic to handle the dust emissions
-  my ($opts, $nl_flags, $definition, $defaults, $nl) = @_;
+  # Logic to handle the dust emissions namelists, both drv_flds_in and lnd_in files
+  my ($opts, $nl_flags, $definition, $defaults, $nl, $envxml_ref) = @_;
 
-  # First get the dust emission method
-  my $var = "dust_emis_method";
-  add_default($opts,  $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, $var );
-
-  my $dust_emis_method = remove_leading_and_trailing_quotes( $nl->get_value($var) );
-
-  my @zender_files_in_lnd_opts = ( "stream_fldfilename_zendersoilerod", "stream_meshfile_zendersoilerod",
-                                   "zendersoilerod_mapalgo" );
-  if ( $dust_emis_method eq "Zender_2003" ) {
-     # get the zender_soil_erod_source
-     add_default($opts,  $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl,
-                 "zender_soil_erod_source", 'dust_emis_method'=>$dust_emis_method );
-
-     my $zender_source = remove_leading_and_trailing_quotes( $nl->get_value('zender_soil_erod_source') );
-     if ( $zender_source eq "lnd" ) {
-        foreach my $option ( @zender_files_in_lnd_opts ) {
-           add_default($opts,  $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, $option,
-                       'dust_emis_method'=>$dust_emis_method, 'zender_soil_erod_source'=>$zender_source,
-                       'hgrid'=>$nl_flags->{'res'}, 'lnd_tuning_mod'=>$nl_flags->{'lnd_tuning_mode'} );
-        }
-     } else {
-        foreach my $option ( @zender_files_in_lnd_opts ) {
-           if ( defined($nl->get_value($option)) ) {
-             $log->fatal_error("zender_soil_erod_source is NOT lnd, but the file option $option is being set" .
-                               " and should NOT be unless you want it handled here in the LAND model, " .
-                               "otherwise the equivalent option is set in CAM" );
-           }
-        }
-     }
+  # Only set dust emission settings -- if not connected to CAM
+  my $lnd_sets_dust = logical_to_fortran($envxml_ref->{'LND_SETS_DUST_EMIS_DRV_FLDS'});
+  if ( &value_is_true( $lnd_sets_dust)) {
+
+      # First get the dust emission method
+      my $var = "dust_emis_method";
+      add_default($opts,  $nl_flags->{'inputdata_rootdir'}, $definition, $defaults, $nl, $var );
+      my $dust_emis_method = remove_leading_and_trailing_quotes( $nl->get_value($var) );
+
+      my @zender_files_in_lnd_opts = ( "stream_fldfilename_zendersoilerod", "stream_meshfile_zendersoilerod",
```

Comment:
Make sure the logic is such that this isn't set if dust_emis_method != Zender_2003

Reply:
I couldn't tell from the diff if the logic is done correctly, but looking at the code it looks fine. So resolving.